docstore-metrics/src/main/java/org/hypertrace/core/serviceframework/docstore/metrics/DocStoreMetricsRegistry.java

```diff
@@ -0,0 +1,117 @@
+package org.hypertrace.core.serviceframework.docstore.metrics;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicLong;
+import org.hypertrace.core.documentstore.Datastore;
+import org.hypertrace.core.documentstore.metric.DocStoreMetric;
+import org.hypertrace.core.documentstore.metric.DocStoreMetricProvider;
+import org.hypertrace.core.documentstore.model.config.CustomMetricConfig;
+import org.hypertrace.core.serviceframework.metrics.PlatformMetricsRegistry;
+
+@SuppressWarnings("unused")
+public class DocStoreMetricsRegistry {
+  private static final long initialDelaySeconds = MINUTES.toSeconds(5);
+
+  /**
+   * To continuously monitor a database and periodically report (standard and custom) metrics.
+   *
+   * <p>The standard metrics (like the database connection count this service is holding) are
+   * reported immediately after this method is invoked and subsequently reported once in every 24
+   * hours.
+   *
+   * <p>The custom metrics are reported at the interval scheduled per metric.
+   *
+   * @param datastore The datastore to be monitored
+   * @param reportingConfigs The custom metric configurations to be reported, if any
+   * @param threadPoolSize The number of threads to be allocated for scheduling. Can just be 1 in
+   *     most cases
+   * @return The created scheduled executor pool
+   */
+  public static ScheduledExecutorService monitor(
```

Comment:
nits:
- try to avoid statics
- thread pool size, custom metrics, etc - let's try to have defaults to make this easier to use
- maybe provide a way of hooking this to platform lifecycle to shutdown executor on shutdown since that's likely how it's going to be used 99% of the time
- a builder type pattern (either a separate builder or just configuring an instance) may make this easier to use

so maybe something like 
```java
DocStoreMetricsRegistry registry = new DocStoreMetricsRegistry(dataStore)
   .withLifecycle(platformLifecycle) // optional
   .withThreadPoolSize(...) // if needed
   .withCustomMetrics(...) // if needed
   .monitor(); // fluent

registry.report(customMetric);

registry.shutdown(); // not needed to be called if given a lifecycle, but can hide the executor
```

What do you think? Main goal I'm going for is ease of use

Reply:
Sure, yeah. That makes sense. Will update.